dpnp/tests/test_product.py

```diff
@@ -593,6 +594,906 @@ def test_order(self, order):
         assert_dtype_allclose(result, expected)
 
 
+class TestMatmul:
```

Comment:
We need also to update `new_umaths_numpy_20` list in `dpnp/tests/test_umath.py`, because `matvec` and `vecmat` are supported now.